chainer/_runtime_info.py

```diff
@@ -0,0 +1,45 @@
+import sys
+
+import numpy
+import six
+
+import chainer
+from chainer.backends import cuda
+
+
+class _RuntimeInfo(object):
+
+    chainer_version = None
+    numpy_version = None
+    cuda_info = None
+
+    def __init__(self):
+        self.chainer_version = chainer.__version__
+        self.numpy_version = numpy.__version__
+        if cuda.available:
+            self.cuda_info = cuda.cupyx.get_runtime_info()
+        else:
+            self.cuda_info = None
+
+    def __str__(self):
+        s = six.StringIO()
+        s.write('''Chainer: {}\n'''.format(self.chainer_version))
+        s.write('''NumPy: {}\n'''.format(self.numpy_version))
+        if self.cuda_info is None:
+            s.write('''CuPy: Not Available\n''')
+        else:
+            s.write('''CuPy:\n''')
+            for line in str(self.cuda_info).splitlines():
+                s.write('''  {}\n'''.format(line))
+        return s.getvalue()
+
+
+def get_runtime_info():
+    return _RuntimeInfo()
+
+
+def print_runtime_info(out=None):
+    if out is None:
+        out = sys.stdout
+    out.write(str(get_runtime_info()))
+    out.flush()
```

Comment:
`flush` method may not exist.
It's better to either avoid it or test its existence first.

Reply:
There's no official specification of file-like objects.
You can define an object with only `write` method and call it a file-like.

Reply:
OK, I fixed to first check existence of the flush method.
(It seems other features (`CupyMemoryProfileHook`, `ProgressBar`, etc...) also assumes `flush`.)

Reply:
I see. It's better to fix them as well.